### ISSUE TITLE
BUG: sys.modules['sys'] not pointing __import__('sys') 

### ISSUE THREAD

**Issue by pmp-p** [micropython/micropython#4923]
Probably the same reason as :
https://github.com/micropython/micropython/issues/4427

**Comment by dpgeorge** [comment]
This is because built-in modules are not registered in `sys.modules`

**Comment by pmp-p** [comment]
ok then, but as they should, is it fixable at the point where import should could call __import__ instead of builtins__import  ?

**Comment by dpgeorge** [comment]
> is it fixable at the point where import should could call `__import__`

Not really.  It can be implemented in the actual import implementation.  But if `__import__` could be overridden then the custom version of import could populate `sys.modules` itself.

**Comment by pmp-p** [comment]
> But if __import__ could be overridden then the custom version of import could populate sys.modules itself.

i think override could not deal with ```__main__``` at startup, before hooking. eg:
```
import sys
sys.modules.get('__main__') # <= nope 
````
so it should be implemented in the actual import implementation
